### PR TITLE
Support for amphora provider services

### PR DESCRIPTION
**Pull request by beagles**
Adds support for launching the amphora controller
services. Implementation is informed by the fact that the amphora provider specific services share many common configurations and deployment requirements.

Note: full deployment support isn't available with this change and would require several manual steps to enable.

### CODE REVIEW

File: api/v1beta1/amphoracontroller_types.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	DatabaseHostname string `json:"databaseHostname,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=octavia
+	// ServiceUser - service user name (TODO: beagles, do we need this at all)
+	ServiceUser string `json:"serviceUser"`
+
+	// +kubebuilder:validation:Required
+	// ServiceAccount - service account name used internally to provide Octavia services the default SA name
+    ServiceAccount string `json:"serviceAccount"`
```
Comment (weinimo) [inline api/v1beta1/amphoracontroller_types.go:51]: Indentation. Probably a tab vs. space issue?

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -12,5 +12,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: quay.io/openstack-k8s-operators/octavia-operator
-  newTag: latest
+  newName: quay.io/beagles/octavia-operator
```
Comment (weinimo) [inline config/manager/kustomization.yaml:15]: I guess this was added accidentally.

**Review by weinimo — CHANGES_REQUESTED** [review]
I found a couple of things while looking at the code. I didn't test it yet.

File: config/manager/kustomization.yaml (hunk 2 above)
Comment (beagles) [reply config/manager/kustomization.yaml:15]: yup, will fix

**Comment by beagles** [comment]
Seems to be having a problem getting the transporturl from rabbit, investigating

**Comment by luis5tb** [comment]
what about the driver-agent component? is that missed? or part of a follow up?

**Comment by beagles** [comment]
> what about the driver-agent component? is that missed? or part of a follow up?

The driver agent is deployed along with the API in a single pod so is already there. The API service and the driver agent are inextricably linked by a socket and are 1:1 relationship so it just made more sense to collocate them in the same pod.

**Comment by luis5tb** [comment]
> > what about the driver-agent component? is that missed? or part of a follow up?
> 
> The driver agent is deployed along with the API in a single pod so is already there. The API service and the driver agent are inextricably linked by a socket and are 1:1 relationship so it just made more sense to collocate them in the same pod.

awesome! thanks!

**Review by dprince — APPROVED** [review]

**Pull request merged**